enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,268 @@
+
+#### Command Line Interface
+
+The command line interface will remain exactly the same as the original `run bundle` command.
+
+#### Assumptions
+
+Like the existing run bundle command, we **assume** the following:
+
+ 1. OLM is already installed
+ 1. kube API is available with the permissions to create resources
+ 1. kubelet can see and pull the provided images (i.e. bundle and index images)
+ 1. If an index image is provided, we assume that it has:
+    - bash, so we can change the entrypoint
+    - if it is a sqlite index image, we assume that:
+        - the existing registry DB located at `/database/index.db`
+        - dependencies required for `opm registry add` so we can inject the bundle into the DB
+
+#### What happens when the `run bundle` command is invoked?
+
+1. The labels from the bundle annotations are parsed. The annotations are received from the bundle image that is passed in as a required argument for the run bundle command. Operator SDK will parse information like channel name and CSV name from the labels in the bundle annotations.
+
+2. The Operator SDK, using the data gathered from the above step as parameters, will invoke Operator Registry APIs that will generate three major components in the form of Operator Registry-defined objects - Package, Channel & Bundle.
+
+3. The Operator SDK will convert these objects into more commonly-consumable formats like JSON/YAML blobs and combine these blobs together to generate a basic file-based catalog representation of the bundle image. In other words, the simplest file-based catalog consists of one package blob, one channel blob, and one bundle blob. These blobs represent the bundle data.
+
+4. If an index image was passed in as a parameter through the `--index-image` flag of the run bundle command, a series of additional steps take place:
+
+    - The index image will be rendered into a declarative config object, similar to the bundle image.
+    - Operator SDK will check if the bundle image is already present in the index image.
+        - If the bundle image is already present in the index image, Operator SDK will not add the bundle to the index but rather error out.
+        - If the bundle is not present in the index image, Operator SDK will add the contents of the bundle image to the existing index image and serve the resulting index image (with the bundle added to it).
+
+    - Note: If the index image that was passed is in SQLite format, Operator SDK will use a different container creation command that uses `opm registry add` to add the bundle to the index and `opm registry serve` to serve the index which is a database file on created on the pod.
```

Comment:
Why we do not use OPM migrate or OPM render to convert any SQLIndex for FBC format so that we can ensure that the SDK implementation will work in the same way in both cases? In this way, we do not need to have logic for each case scenario.

Reply:
Answered this in [comment](https://github.com/operator-framework/enhancements/pull/112#discussion_r856350119)